Classes/Validation/Common/DomNodeListValidator.php

```diff
@@ -122,7 +116,7 @@ public function validateHasAny(): DomNodeListValidator
     public function validateHasOne(): DomNodeListValidator
     {
         if ($this->nodeList->length != 1) {
-            $this->addError('There must be an element', 1736504354);
+            $this->addMessage('There must be no more than one element', 1736504354);
```

Comment:
The message does not match the condition in the `if` statement.

Reply:
Why? The test checks whether the list contains exactly one element. If not, a message is added to the result.

Reply:
Zero is also "no more than one". Just use the text from you answer: "There must be exactly one element".